packages/kbn-reporting/public/share/share_context_menu/register_csv_modal_reporting.tsx

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { i18n } from '@kbn/i18n';
+import React from 'react';
+import { toMountPoint } from '@kbn/kibana-react-plugin/public';
+
+import { CSV_JOB_TYPE, CSV_JOB_TYPE_V2 } from '@kbn/reporting-export-types-csv-common';
+
+import type { SearchSourceFields } from '@kbn/data-plugin/common';
+import { ShareContext, ShareMenuItem, ShareMenuProvider } from '@kbn/share-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import url from 'url';
+import type { ExportModalShareOpts } from '.';
+import { checkLicense } from '../..';
+
+export const reportingCsvShareProvider = ({
+  apiClient,
+  application,
+  license,
+  usesUiCapabilities,
+  theme,
```

Comment:
You will need an `I18nStart` object for the "new" `toMountPoint` utility
```suggestion
  i18n: i18nStart,
  theme,
```